src/tools/debugger/session/Breakpoints.java

```diff
@@ -91,20 +108,42 @@ public Breakpoint addOrUpdate(final AsyncMessageReceiveBreakpoint bId) {
       truffleBreakpoints.put(bId, bp);
     }
     bp.setEnabled(bId.isEnabled());
-    return bp;
   }
 
-  public synchronized void addOrUpdate(final MessageReceiveBreakpoint bId) {
+  public synchronized void addOrUpdate(final MessageReceiverBreakpoint bId) {
     FullSourceCoordinate coord = bId.getCoordinate();
-    BreakpointEnabling<MessageReceiveBreakpoint> existingBP = receiverBreakpoints.get(coord);
+    BreakpointEnabling<MessageReceiverBreakpoint> existingBP = receiverBreakpoints.get(coord);
     if (existingBP == null) {
-      existingBP = new BreakpointEnabling<MessageReceiveBreakpoint>(bId);
+      existingBP = new BreakpointEnabling<MessageReceiverBreakpoint>(bId);
       receiverBreakpoints.put(coord, existingBP);
     } else {
       existingBP.setEnabled(bId.isEnabled());
     }
   }
 
+  public synchronized void addOrUpdate(final PromiseResolverBreakpoint bId) {
```

Comment:
I refactored the addOrUpdate for MessageSenderBreakpoint and AsyncMessageReceiverBreakpoint...But for the case of MessageReceiverBreakpoint and the promises breakpoint I have been trying to do something like this:

` public synchronized void addOrUpdate(final PromiseResolutionBreakpoint bId) {
    saveBreakpointsEnabling(bId, promiseResolutionBreakpoints);
  }

  private void saveBreakpointsEnabling(final SectionBreakpoint bId, final Map<FullSourceCoordinate, BreakpointEnabling<? extends SectionBreakpoint>> map) {
    FullSourceCoordinate coord = bId.getCoordinate();
    BreakpointEnabling<? extends SectionBreakpoint> existingBP = map.get(coord);
    if (existingBP == null) {
      existingBP = new BreakpointEnabling<>(bId);
      map.put(coord, existingBP);
    } else {
      existingBP.setEnabled(bId.isEnabled());
    }
  }`

the problem is that the argument types are not applicable for '? extends SectionBreakpoint', I'm not sure if there is a way of refactoring these three methods. The only difference between them is the map (receiverBreakpoints, promiseResolverBreakpoints, promiseResolutionBreakpoints)